coding-concepts.js

```diff
@@ -10,34 +10,54 @@
 
 // --------------------2) What will this log?
 
-const greeter = (name) => {
-  `Hello, ${name}!`
-}
+// const greeter = (name) => {
+  // `Hello, ${name}!`
+// }
 // console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: "Hello, LEARN Student!"
+// b) Verify and explain: Dang it, you got me! There's no return on line 20 so when we called on our function we got 'undefined'. greeter is still preforming its function, but it isn't returning the output to us when we call on it. If we wanted to fix this, we would simply need to add 'return' on line 20 in front of `Hello, ${name}!`. See below!
+
+// const greeter = (name) => {
+//   return `Hello, ${name}!`
+// }
+// console.log(greeter("LEARN Student"))
+
+// Alternatively, if we don't want to explicitly code a return, and want to have our function on just one line, we can remove the {} establishing our code block, and instead wrap our entire function after the assignment operator in (). See below!
+
+const greeter = ((name) => `Hello, ${name}!`)
+console.log(greeter("LEARN Student"))
+
 
 // --------------------3) What will this log?
 
 const multipliedByTwo = [4, 5, 6, 7, 8].map((number) => number * 2)
 // console.log(multipliedByTwo)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: We should receive an array that is the same length as our original array, except each index in our new array will be the original value multiplied by two.
+
+// b) Verify and explain: output: [ 8, 10, 12, 14, 16 ] 
+
+// Perfect! .map() is a higher order function that we can use within another function to iterate through each value in an array, mutate the data, and then return a new array of the same length with the new values. In our example, our .map() is taking each (number) in our original array and multiplying it by two, then returning a new array with those numbers. 
 
 // --------------------4) What will this log?
 
 const onlyOdds = [11, 12, 13, 14, 15].filter((number) => number % 2 !== 0)
 // console.log(onlyOdds)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This should log a new array that consists of ONLY the numbers after being divided by two have a remainder other than 0, which indicates an odd number. So, more simply, an array of only odd numbers from out original array. [ 11, 13, 15 ]
+
+// b) Verify and explain: output: [ 11, 13, 15 ]
+
+// Cool! So .filter() is a higher order function that can be used within a function to iterate through an arrays values and output a new array of only values that match what criteria you want to apply. .filter() has a built in 'if' statement that can be used to return true statements. So in our example we are basically saying 'if a number in our array divided by 2 has a remainder of anything less than zero, add that number to a new array' which returns an array of only odd numbers.
```

Comment:
Exactly, filter is extremely useful for those situations where we want to check for a condition.